[Bug Report] make check-ci fails one test on CUDA >= 10.2

**Describe the bug**
I ran `make check-ci` on an AWS EC2 instance and one test fails:

`make check-ci`

```
>           return _VF.einsum(equation, operands)  # type: ignore[attr-defined]
E           RuntimeError: Deterministic behavior was enabled with either `torch.use_deterministic_algorithms(True)` or `at::Context::setDeterministicAlgorithms(true)`, but this operation is not deterministic because it uses CuBLAS and you have CUDA >= 10.2. To enable deterministic behavior in this case, you must set an environment variable before running your PyTorch application: CUBLAS_WORKSPACE_CONFIG=:4096:8 or CUBLAS_WORKSPACE_CONFIG=:16:8. For more information, go to https://docs.nvidia.com/cuda/cublas/index.html#results-reproducibility

../.cache/pypoetry/virtualenvs/sae-lens-ODzxvHul-py3.10/lib/python3.10/site-packages/torch/functional.py:385: RuntimeError

FAILED tests/unit/training/test_cache_activations_runner.py::test_cache_activations_runner - RuntimeError: Deterministic behavior was enabled with either `torch.use_deterministic_algorithms(True)` or `at::Context::setDeterministicAlgorithm...
================================================ 1 failed, 94 passed, 4 skipped, 19 warnings in 46.62s ================================================
```

I followed its suggestion and it succeeds:
`CUBLAS_WORKSPACE_CONFIG=:4096:8 make check-ci`
Makes all tests pass

Not sure if there is something to fix here, but Joseph asked me to make sure make check-ci passes on the new workflow I'm building.

**System Info**
Ubuntu on AWS EC2
AMI: Deep Learning OSS Nvidia Driver AMI GPU PyTorch 2.2.0 (Ubuntu 20.04) 20240507
AMI ID: ami-02a07d31009cc8717


### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/jbloomAus/SAELens/issues) in the repo (**required**)

--- a/tests/unit/conftest.py
+++ b/tests/unit/conftest.py
@@ -17,7 +17,6 @@ def reproducibility():
     seed = 0x1234_5678_9ABC_DEF0
     torch.manual_seed(seed)
     torch.cuda.manual_seed_all(seed)
-    torch.use_deterministic_algorithms(True)
     torch.backends.cudnn.benchmark = False
     # Python native RNG; docs don't give any limitations on seed range
     random.seed(seed)

